charts/firefly/values.yaml

```diff
@@ -28,31 +28,37 @@ jupyterhub:
         kubespawner_override:
           image: "ghcr.io/gohypergiant/hypergiant-jupyter:cpu-huggingface-pytorch"
           http_timeout: 3600
+          image_pull_policy: "Always"
       - display_name: "Spark (CPU)"
         description: "Apache Spark, CPU only."
         kubespawner_override:
           image: "ghcr.io/gohypergiant/hypergiant-jupyter:cpu-spark"
           http_timeout: 3600
+          image_pull_policy: "Always"
       - display_name: "Pytorch (CPU)"
         description: "Pytorch, CPU only."
         kubespawner_override:
           image: "ghcr.io/gohypergiant/hypergiant-jupyter:cpu-pytorch"
           http_timeout: 3600
+          image_pull_policy: "Always"
       - display_name: "XGBoost (CPU)"
         description: "XGBoost, CPU only."
         kubespawner_override:
           image: "ghcr.io/gohypergiant/hypergiant-jupyter:cpu-xgboost"
           http_timeout: 3600
+          image_pull_policy: "Always"
       - display_name: "Tensorflow (CPU)"
         description: "Tensorflow, CPU only."
         kubespawner_override:
           image: "ghcr.io/gohypergiant/hypergiant-jupyter:cpu-tensorflow"
           http_timeout: 3600
+          image_pull_policy: "Always"
       - display_name: "Hugging Face + Tensorflow (CPU)"
         description: "Hugging Face and Tensorflow, CPU only."
         kubespawner_override:
           image: "ghcr.io/gohypergiant/hypergiant-jupyter:cpu-huggingface-tensorflow"
           http_timeout: 3600
+          image_pull_policy: "Always"
```

Comment:
why are we setting pull policy always? are tags for containers getting re-used? This would also negate the image pre-puller feature of jupyyterhub

Reply:
right now, yeah unfortunately these image tags are being re-used.